pkg/scheduler/backend/queue/scheduling_queue.go

```diff
@@ -599,7 +618,14 @@ func (p *PriorityQueue) Activate(logger klog.Logger, pods map[string]*v1.Pod) {
 	for _, pod := range pods {
 		if p.activate(logger, pod) {
 			activated = true
+			continue
 		}
+
+		// If this pod is in-flight, register the activation event or update moveRequestCycle so that the pod will be requeued when it comes back.
```

Comment:
"or update moveRequestCycle (when QHints are disabled)"

BTW. I was wondering how we record the events and it's actually quite clever. I'd however rename it from inFlightEvents to inFlightPodsAndEvents, as I haven't suspected that different objects are mixed.

Reply:
Yeah, "inFlightEvents (hey, Pods are there too!)" is a bit confusing 😅